sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -74,13 +74,71 @@ object RDDConversions {
   }
 }
 
+private[sql] object ExternalRDD {
+
+  def apply[T: Encoder](rdd: RDD[T])(session: SparkSession): LogicalPlan = {
```

Comment:
Actually I'm not sure why we curry the constructor either. Since `RDDScan` does it, it's ok we follow it. But for this apply `method`, I don't see the value of doing it. cc @yhuai